conversions/test/Test/Data/Conversions/Integral.hs

```diff
@@ -0,0 +1,265 @@
+    , integerToIntegral
+    , naturalToWord
+    ]
+
+integersToNatural :: Tasty.TestTree
+integersToNatural
+  = Tasty.testGroup "Integers to Natural conversions"
+    [ toNatural Hedgehog.Gen.int
+    , toNatural Hedgehog.Gen.int8
+    , toNatural Hedgehog.Gen.int16
+    , toNatural Hedgehog.Gen.int32
+    , toNatural Hedgehog.Gen.int64
+    , integerToNatural
+    ]
+  where
+    toNatural
+      :: forall integer
+      .  ( Bounded integer
+         , Conversion (Result Natural integer) Natural
+         , Conversion (Result integer Natural) integer
+         , Num integer
+         , Show integer
+         , Typeable integer
+         )
+      => (Hedgehog.Range integer -> Hedgehog.Gen integer)
+      -> Tasty.TestTree
+    toNatural genInteger
+      = Test.run biConvert generators
+      where
+        generators :: Test.Generators integer
+        generators = Test.Generators
+          { success = [genInteger range]
+          , failure = [pure $ minBound @integer]
+          }
+          where
+            range :: Hedgehog.Range integer
+            range = Hedgehog.Range.constant 0 maxBound
+
+        biConvert :: Test.BiConvert Natural integer
+        biConvert = Test.BiConvert
+          { toTarget   = Test.mkConvertEither $ convert @(Result integer Natural)
+          , fromTarget = Test.mkConvertEither $ convert @(Result Natural integer)
+          }
+
+    integerToNatural :: Tasty.TestTree
+    integerToNatural
+      = Test.run biConvert generators
+      where
+        biConvert :: Test.BiConvert Natural Integer
+        biConvert = Test.BiConvert
+          { toTarget   = Test.mkConvertEither $ convert @(Either (UserBoundError Integer Text) Natural)
+          , fromTarget = pure . convert
+          }
+
+        generators :: Test.Generators Integer
+        generators = Test.Generators
+          { success = [Hedgehog.Gen.integral range]
+          , failure = [pure . convert $ minBound @Int64]
+          }
+          where
+            range :: Hedgehog.Range Integer
+            range = Hedgehog.Range.constant 0 (succ . convert $ maxBound @Int64)
+
+
+naturalToWord :: Tasty.TestTree
+naturalToWord
+  = Tasty.testGroup "Natural to Word conversions"
+    [ toWord Hedgehog.Gen.word8
+    , toWord Hedgehog.Gen.word16
+    , toWord Hedgehog.Gen.word32
+    , run @Word64 generatorsWord64
+    -- Word & Word64 have the same precision
+    , run @Word generatorsWord64
+    ]
+  where
+    toWord
+      :: forall word
+      .  ( Bounded word
+         , Conversion Natural word
+         , Conversion (Result Natural word) Natural
+         , Show word
+         , Typeable word
+         )
+      => (Hedgehog.Range word -> Hedgehog.Gen word)
+      -> Tasty.TestTree
+    toWord genSource
+      = run @word generators
+      where
+        generators :: Test.Generators Natural
+        generators = Test.Generators
+          { success = [convert <$> genSource range]
+          , failure = [pure (convert $ maxBound @Word64)]
```

Comment:
__suggestion:__ Sort of along the same lines as my [previous statement](https://github.com/mbj/mhs/pull/243/files#r1231447485) could we specify a range that is *just* above the boundary up to `maxBound @Word64`?

Reply:
I made your suggested change which allowed me to eliminate a standalone test for `Word64`